Composer/packages/client/src/pages/setting/SettingsPage.tsx

```diff
@@ -50,6 +52,14 @@ const SettingPage: React.FC<RouteComponentProps<{ '*': string }>> = () => {
   const { defaultLanguage, languages } = useRecoilValue(settingsState);
   const { navigate } = useLocation();
 
+  // when fresh page, projectId in store are empty, no project are opened at client
+  // use cached projectId do fetch.
+  const cachedProjectId = useProjectIdCache();
+  useEffect(() => {
+    if (!projectId && cachedProjectId) {
+      fetchProjectById(cachedProjectId);
+    }
+  }, [projectId]);
```

Comment:
since it's duplicated, we can move it inside the shared hook.

Reply:
I think it's not been used in a lot place or complex enough. maybe leave it here would be more flexible.

Reply:
If the fix only applies to the page refresh scenario, the deps of `useEffect` can become `[]`?

Reply:
I moved fetch project logic to root router, better than do this in components.